hackathon_site/dashboard/frontend/src/components/user/AcknowledgementForm/AcknowledgementForm.js

```diff
@@ -0,0 +1,180 @@
+        //     />
+        //     <FormControlLabel
+        //         control={<Checkbox name="forthCheck" />}
+        //         label={checkBox4}
+        //     />
+        //     <Typography variant="body1" className={styles.acknowledgeText}>
+        //         {firstHalf}
+        //         <Link href="#" onClick={openDoc}>
+        //             Waiver of Liability and Hold Harmless Agreement
+        //         </Link>
+        //         {secondHalf}
+        //     </Typography>
+        //     <TextField
+        //         label="e-signature"
+        //         className={styles.acknowledgeSignature}
+        //         name="signature"
+        //     />
+        //     <Button
+        //         type="submit"
+        //         className={styles.acknowledgeButton}
+        //         variant="contained"
+        //         color="primary"
+        //         onClick={handleContinue}
+        //     >
+        //         Continue
+        //     </Button>
+        // </form>
+        <div className={styles.acknowledge} onSubmit={handleContinue}>
+            <FormikCheckBox
+                name="agreement"
+                value={checkBox1}
+                className={styles.acknowledgeCheckbox}
+            />
+            <FormikCheckBox
+                name="agreement"
+                value={checkBox2}
+                className={styles.acknowledgeCheckbox}
+            />
+            <FormikCheckBox
+                name="agreement"
+                value={checkBox3}
+                className={styles.acknowledgeCheckbox}
+            />
+            <FormikCheckBox
+                name="agreement"
+                value={checkBox4}
+                className={styles.acknowledgeCheckbox}
+            />
+            <Typography variant="body1" className={styles.acknowledgeText}>
+                {firstHalf}
+                <Link href="#" onClick={openDoc}>
+                    Waiver of Liability and Hold Harmless Agreement
+                </Link>
+                {secondHalf}
+            </Typography>
+            <TextField
+                label="e-signature"
+                className={styles.acknowledgeSignature}
+                name="signature"
+            />
+            <Button
+                type="submit"
+                className={styles.acknowledgeButton}
+                variant="contained"
+                color="primary"
+                onClick={handleContinue}
+            >
+                Continue
+            </Button>
+        </div>
+    );
+};
+
+export const EnhancedAcknowledgmentForm = ({
+    handleSubmit,
+    requestFailure,
+    openDoc,
+}) => {
+    const onSubmit = (formikValues) => {
+        handleSubmit({ signature: formikValues.signature });
+    };
+
+    return (
+        <Formik
+            initialValues={{ signature: "", agreement: [""] }}
+            onSubmit={onSubmit}
+            validateOnBlur={false}
+            validateOnChange={false}
```

Comment:
You should add a `validationSchema` in here as well, which you build using `Yup`. Take a look at the one used by the login form: https://github.com/ieeeuoft/hackathon-template/blob/develop/hackathon_site/dashboard/frontend/src/components/user/LoginForm/LoginForm.js#L21

For you, each check box would have a requirement like `Yup.boolean().required(errorMessage)`, and the e-signature box would be `Yup.string().required(errorMessage)`. That way they can't submit the form until all fields have been filled out (and you should have an acompanying test for that, of course).